eth/api.go

```diff
@@ -812,3 +813,23 @@ func (api *ScrollAPI) GetSkippedTransactionHashes(ctx context.Context, from uint
 
 	return hashes, nil
 }
+
+// CalculateRowConsumptionByBlockNumber
+func (api *ScrollAPI) CalculateRowConsumptionByBlockNumber(ctx context.Context, number rpc.BlockNumber) (*types.RowConsumption, error) {
+	block := api.eth.blockchain.GetBlockByNumber(uint64(number.Int64()))
+	if block == nil {
+		return nil, errors.New("block not found")
+	}
+
+	// todo: fix temp AsyncChecker leaking the internal Checker instances
```

Comment:
What do you mean by "leaking" the internal instances? Is it just that we create new ones but never deallocate them?

Reply:
Yeap, exactly. The allocation that happens on rust side is never deallocated. Becuase there is no way to do that.